src/kern/shared.h

```diff
@@ -14,16 +15,18 @@ typedef struct
 typedef struct
 {
     unsigned long syscallid;
+    int pid;
```

Comment:
I guess this is good idea to create new struct as "common" and reuse it in the other structs. Furthermore, it is a prevalent way.

```suggestion
    typedef struct
{
    unsigned long syscallid;
    int pid;
} __attribute__((aligned(8))) struct_common;
```

Reply:
It is a very good idea